src/Adapter/Address/AbstractAddressHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Address;
+
+use Address;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Address\ValueObject\AddressId;
+
+/**
+ * Provides reusable methods for address command/query handlers
+ */
+abstract class AbstractAddressHandler
+{
+    /**
+     * Validates that requested address was found
+     *
+     * @param AddressId $addressId
+     * @param Address $address
+     *
+     * @throws AddressNotFoundException
+     */
+    protected function assertAddressWasFound(AddressId $addressId, Address $address)
+    {
+        if ($address->id !== $addressId->getValue()) {
```

Comment:
Does both have strict types? I think in  `Address` entity by default  `id` field is string so this code part can lead to unexpected behavior in some cases. You can type cast it or just check if its more then zero `0 >= $address->id`.

Reply:
Legacy object model is built from AddressId which already verifies id value to be integer in this case.

Reply:
I'm hesitating here 😅 yes AddressId ensures it will be an integer on the right ... however
- if `$address->id` isn't it a bad thing ? shouldn't we check that, even if it will be checked indirectly ?
- what if AddressId definition changes in the future ? I'm thinking about some defensive programming here

Reply:
> if $address->id isn't it a bad thing

What do you mean?

> what if AddressId definition changes in the future ?

It means BC break and all the clients should be updated. 

It does not look like an issue though.